docs/adr/0001-pr-polling-workflow.md

```diff
@@ -0,0 +1,55 @@
+# 1. Pull Request Polling Workflow
+
+* Status: proposed
+* Date: 2023-06-20 
+* Authors: @yitsushi
+* Deciders: @yitsushi @chanwit @squaremo @yiannistri
+
+## Context
+
+To detect pull request changes, we can use webhooks or polling using GitHub's
+API.
+
+## Decision
+
+We decided to use polling for security reasons. Using webhook would require to
+open an ingress to the cluster and that's not acceptable especially in an
```

Comment:
I don't find this universally-quantified statement very convincing. My estimation, coming from Flux, is that plenty of people are happy to use webhooks. I am fine with the _decision_, though. A better way to argue it would be that a significant proportion of people will not want to use webhooks, but polling will nearly always work, so it's better to start with polling.

Reply:
> A better way to argue it would be that a significant proportion of people will not want to use webhooks

This is a decision record, it's not an RFC discussing where to go. It just records that we decided to use polling and the main reason was security and air-gapped environments.

Reply:
OK, so the correct place to register my objection to this line of reasoning is the RFC; I'll do that.

This makes a material difference, just in case that's in question -- if you read this rationale and took it at face value, you might think "we should never implement webhooks"; but I think that would be a poor conclusion to draw. Reasons matter.